src/azure-cli/azure/cli/command_modules/cosmosdb/aaz/latest/cosmosdb/postgres/cluster/_create.py

```diff
@@ -0,0 +1,595 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "cosmosdb postgres cluster create",
+)
+class Create(AAZCommand):
+    """Create a new cluster with servers.
+    """
+
+    _aaz_info = {
+        "version": "2022-11-08",
+        "resources": [
+            ["mgmt-plane", "/subscriptions/{}/resourcegroups/{}/providers/microsoft.dbforpostgresql/servergroupsv2/{}", "2022-11-08"],
+        ]
+    }
+
+    AZ_SUPPORT_NO_WAIT = True
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        return self.build_lro_poller(self._execute_operations, self._output)
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.cluster_name = AAZStrArg(
+            options=["-n", "--name", "--cluster-name"],
+            help="The name of the cluster.",
+            required=True,
+            fmt=AAZStrArgFormat(
+                pattern="^(?![0-9]+$)(?!-)[a-z0-9-]{3,40}(?<!-)$",
+                max_length=40,
+                min_length=3,
+            ),
```

Comment:
Echo the above comment, once you defined the validations for the `name` of `check-name-availability` command, it will generate format like this one.